core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitUtils.java

```diff
@@ -232,4 +238,47 @@ public static boolean isAfterImageColumn(String columnName, TableMetadata tableM
     }
     return !isBeforeImageColumn(columnName, tableMetadata);
   }
+
+  static Put createPutForInsert(Insert insert) {
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(insert.forNamespace().orElse(null))
+            .table(insert.forTable().orElse(null))
+            .partitionKey(insert.getPartitionKey());
+    insert.getClusteringKey().ifPresent(buildable::clusteringKey);
+    insert.getColumns().values().forEach(buildable::value);
+    buildable.enableInsertMode();
+    return buildable.build();
+  }
+
+  static Put createPutForUpsert(Upsert upsert) {
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(upsert.forNamespace().orElse(null))
+            .table(upsert.forTable().orElse(null))
+            .partitionKey(upsert.getPartitionKey());
+    upsert.getClusteringKey().ifPresent(buildable::clusteringKey);
+    upsert.getColumns().values().forEach(buildable::value);
+    buildable.enableImplicitPreRead();
+    return buildable.build();
+  }
+
+  static Put createPutForUpdate(Update update) {
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(update.forNamespace().orElse(null))
+            .table(update.forTable().orElse(null))
+            .partitionKey(update.getPartitionKey());
+    update.getClusteringKey().ifPresent(buildable::clusteringKey);
+    update.getColumns().values().forEach(buildable::value);
+    if (update.getCondition().isPresent()) {
+      update
+          .getCondition()
+          .ifPresent(c -> buildable.condition(ConditionBuilder.putIf(c.getExpressions())));
+    } else {
+      buildable.condition(ConditionBuilder.putIfExists());
+    }
+    buildable.enableImplicitPreRead();
+    return buildable.build();
+  }
```

Comment:
For Update in Consensus Commit, we enable the implicit pre-read, and if no condition is specified in the Update object, we set `putIfExists` for the put operation, or if a condition is specified, we set the same condition for the put operation.

Reply:
> if a condition is specified, we set the same condition for the Put object.

Even w/o `putIfExists`, an update operation [fails](https://github.com/scalar-labs/scalardb/blob/c49d06482361c8e98e0ad5b6472319e066968fd9/core/src/main/java/com/scalar/db/transaction/consensuscommit/MutationConditionsValidator.java#L50) if the target record doesn't exist, right?

Reply:
Yes, that's correct. If a condition is specified, we set the same condition (`putIf`) for the Put object. In this case, if the target record doesn't exist, the update operation fails.